php_phongo.c

```diff
@@ -1355,6 +1356,203 @@ static void php_phongo_free_ssl_opt(mongoc_ssl_opt_t *ssl_opt)
+#endif
+
+	p_event->client = client;
+	p_event->command_name = estrdup(mongoc_apm_command_succeeded_get_command_name(event));
+	p_event->server_id = mongoc_apm_command_succeeded_get_server_id(event);
+	p_event->operation_id = mongoc_apm_command_succeeded_get_operation_id(event);
+	p_event->request_id = mongoc_apm_command_succeeded_get_request_id(event);
+	p_event->duration_micros = mongoc_apm_command_succeeded_get_duration(event);
+	p_event->reply = bson_copy(mongoc_apm_command_succeeded_get_reply(event));
+
+#if PHP_VERSION_ID >= 70000
+	php_phongo_dispatch_handlers("commandSucceeded", &z_event);
+#else
+	php_phongo_dispatch_handlers("commandSucceeded", z_event);
+#endif
+	zval_ptr_dtor(&z_event);
+}
+
+static void php_phongo_command_failed(const mongoc_apm_command_failed_t *event)
+{
+	mongoc_client_t *client = mongoc_apm_command_failed_get_context(event);
+	php_phongo_commandfailedevent_t *p_event;
+#if PHP_VERSION_ID >= 70000
+	zval  z_event;
+#else
+	zval *z_event = NULL;
+#endif
+	bson_error_t tmp_error;
+	zend_class_entry *default_exception_ce;
+	TSRMLS_FETCH();
+
+	default_exception_ce = zend_exception_get_default(TSRMLS_C);
+
+	/* Check for subscriber size */
+	if (zend_hash_num_elements(&MONGODB_G(subscribers)) == 0) {
+		return;
+	}
+
+#if PHP_VERSION_ID >= 70000
+	object_init_ex(&z_event, php_phongo_commandfailedevent_ce);
+	p_event = Z_COMMANDFAILEDEVENT_OBJ_P(&z_event);
+#else
+	MAKE_STD_ZVAL(z_event);
+	object_init_ex(z_event, php_phongo_commandfailedevent_ce);
+	p_event = Z_COMMANDFAILEDEVENT_OBJ_P(z_event);
+#endif
+
+	p_event->client = client;
+	p_event->command_name = estrdup(mongoc_apm_command_failed_get_command_name(event));
+	p_event->server_id = mongoc_apm_command_failed_get_server_id(event);
+	p_event->operation_id = mongoc_apm_command_failed_get_operation_id(event);
+	p_event->request_id = mongoc_apm_command_failed_get_request_id(event);
+	p_event->duration_micros = mongoc_apm_command_failed_get_duration(event);
+
+	/* We need to process and convert the error right here, otherwise
+	 * debug_info will turn into a recursive loop, and with the wrong trace
+	 * locations */
+	mongoc_apm_command_failed_get_error(event, &tmp_error);
+	{
+#if PHP_VERSION_ID < 70000
+		MAKE_STD_ZVAL(p_event->z_error);
+		object_init_ex(p_event->z_error, phongo_exception_from_mongoc_domain(tmp_error.domain, tmp_error.code));
+		zend_update_property_string(default_exception_ce, p_event->z_error, "message", sizeof("message")-1, tmp_error.message TSRMLS_CC);
+		zend_update_property_long(default_exception_ce, p_event->z_error, "code", sizeof("code")-1, tmp_error.code TSRMLS_CC);
+#else
+		object_init_ex(&p_event->z_error, phongo_exception_from_mongoc_domain(tmp_error.domain, tmp_error.code));
+		zend_update_property_string(default_exception_ce, &p_event->z_error, "message", sizeof("message")-1, tmp_error.message TSRMLS_CC);
+		zend_update_property_long(default_exception_ce, &p_event->z_error, "code", sizeof("code")-1, tmp_error.code TSRMLS_CC);
```

Comment:
I copied this from somewhere else though? Perhaps that was before you fixed things up.

The whole point of making the exception here, is that it gets the right trace information. THe original in HHVM showed that it needed to be in this spot as well. Sadly, the overview.phpt test no longer shows it, as you wanted that caught by a ``%a``. We should manually check this.